rust/parquet/benches/codec.rs

```diff
@@ -0,0 +1,196 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#![feature(test)]
+extern crate parquet;
+#[macro_use]
+extern crate lazy_static;
+extern crate test;
+use test::Bencher;
+
+use std::{env, fs::File};
+
+use parquet::{basic::Compression, compression::*, file::reader::*};
+
+// 10k rows written in page v2 with type:
+//
+//   message test {
+//     required binary binary_field,
+//     required int32 int32_field,
+//     required int64 int64_field,
+//     required boolean boolean_field,
+//     required float float_field,
+//     required double double_field,
+//     required fixed_len_byte_array(1024) flba_field,
+//     required int96 int96_field
+//   }
+//
+// filled with random values.
+const TEST_FILE: &str = "10k-v2.parquet";
```

Comment:
I think we normally put test data in https://github.com/apache/parquet-testing so perhaps we should add this one there as well (or if there any existing file there that we can use instead)?

Reply:
I created https://github.com/apache/parquet-testing/pull/15 -- if/when that gets merged in, I'll update this PR to pick up a later version of parquet-testing and remove the binar from this PR as well.